### DISCUSSION TITLE
How are the properties of a pn is calculated BEFORE adding geometric models?

### DISCUSSION THREAD

**Discussion by fsamaraie** [PMEAL/OpenPNM#2707] [Q&A]
Hello community, 
As you know, the when pn=op.io.network_from_porespy(snow_output.network) is ran, we get a pn that already has alot of properties, including equivalent_diameter and inscribed_diameter. My understanding is that there are different models, such as spheres_and_cylinders that can be added to pn for the purpose of modeling pores as spheres or other shapes, however, as you know, some geometric properties are already present before any model is added. I am trying to know how these properties were calculated. I tried to look for the source code of network_from_porespy (https://openpnm.org/_modules/openpnm/io/_porespy.html#network_from_porespy) but it seems that I cannot find these details are explained, as I only get what each input mean. I found a similar discussion here (Seed and Pore Diameter #1932) but the link seems broken. 
Any help is highly appreciated.

**Comment by jgostick** [comment]
The `snow2` function extracts a fairly large number of geometric values from the image, mostly based on the distance transform of the void space.  To see the actually computations you'll need to look at that source code.  The `network_from_porespy` function only reads in the dictionary that comes from `snow2`

Once the dictionary is imported into an OpenPNM network, *then* you have to decide which diameter to I want to be my actual diamter, so do something like ``pn['pore.diameter'] = pn['pore.equivalent_diameter']``.  You notice that `snow2` does not include `'pore.diameter'`...which we do on purpose to force the user to select the one they want.  We used to be opintionated about this, but in V3 we are moving to "let the user make more decisions" approach.  

Then after you have defined what you want to be the acting diameter, OpenPNM will use that for all it's calculations, like pore volume, diffusive conductance, etc.

**Reply by fsamaraie** [reply]
Dear @jgostick 
Many thanks for your response. My confusion is just stemmed from the fact that there are alot of geometric properties that snow2 shows which mean that a given geometric shape is assumed. For example, we have equivalent diameter. It is not clear for me how this diameter was calculated (ie, based on a sphere or cube volume). Through the forum I learnt it is based on a sphere of a similar volume. However if a sphere is already assumed, why would we later add sphere and cylinder model?
Thanks again.

**Comment by likeduck** [comment]
Hope not to disturb you, I have a question about SNOW output too.

I want to calculate the shape factor of the SNOW output, but when I use the following code, it could not be plotted in a figure due to there are inf in the shape facter. Then I print some properties in SNOW output and found that the perimeter of throat has 0 in it.

```
pn['throat.shape_factor'] = np.pi * pn['throat.cross_sectional_area'] / (pn['throat.perimeter'])**2
print(pn['throat.perimeter')
```

![捕获](https://user-images.githubusercontent.com/117707506/231388448-92fff3a8-c09a-43c8-bb2b-f6f5c575f5c5.PNG)

Could you please tell me if this situation health? Is there any way to avoid or improve this situation?

Many Thanks！
